### PR TITLE
Fix disabled state

### PR DESCRIPTION
**Pull request by job13er**
# minor# change to address UX issues w.r.t. disabled/error states as well as refactoring some code.

### CODE REVIEW

**Comment by job13er** [comment]
WARNING: the diffs on the JS code will be horrible. I separated out the refactoring into the first commit, so if you click on the individual commits after that, you can see better the actual changes made.

**Comment by rox163** [comment]
:+1:

**Comment by job13er** [comment]
@rox163 I've updated the spacing, I think it looks good now:

screen shot 2016-03-17 at 2 35 09 pm

**Comment by sophypal** [comment]
:+1:

**Comment by sandersky** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -1,5 +1,3 @@
-[ci-img]: https://img.shields.io/travis/ciena-frost/ember-frost-select.svg "Travis CI Build Status"
-[ci-url]: https://travis-ci.org/ciena-frost/ember-frost-select
```
Comment (job13er) [inline README.md]: Nobody noticed I accidentally removed these lines from the `README.md`? :disappointed:

**Pull request merged**